<?xml version="1.0"?>

<!--
Author: Chris Moneyron, Purdue University, cmoneyron@gmail.com
Professor: Nina Mahmoudian, Purdue University, ninam@purdue.edu

Launch file for swarm of Crazyflies
List each uri as arg and include QTM body name in param 'cf_names' (i.e. CF1)
Remove unused Crazyflie names from cf_names (nodes for non-used CF groups automatically not created)
-->
     
<launch>
  <!-- List all uris (same channel/data rate but different addresses) -->
  <arg name="uri1" default="radio://0/80/2M/0xE7E7E7E701" />
  <arg name="uri2" default="radio://0/80/2M/0xE7E7E7E702" />
  <arg name="uri3" default="radio://0/80/2M/0xE7E7E7E703" />
  <arg name="uri4" default="radio://0/80/2M/0xE7E7E7E704" />
  <arg name="uri5" default="radio://0/80/2M/0xE7E7E7E705" />
  
  <arg name="frame" default="/world" />
  <arg name="ip" default="192.168.1.166" /> <!-- Qualisys DHCP Advanced Tab -->
  <arg name="port" default="22223" />

  <!--<include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>-->
  
  <!-- CF Qualisys body names; string format: "CF1,CF2,CF3,CF4,CF5" -->
  <arg name="cf_names_arg" value="CF4" />
  <param name="cf_names" value="$(arg cf_names_arg)" />

  <!-- Charging Pad Qualisys body names; string format: "Pad1,Pad2,Pad3,Pad4,Pad5" -->
  <param name="pad_names" value="" />

  <!-- Other parameters -->
  <param name="work_width" value="4" />
  <param name="work_length" value="5" />

  <!-- Group/Node for each Crazyflie -->
  <!--<group ns="CF1" if="$(eval 'CF1' in cf_names_arg)">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri1)" />
      <param name="tf_prefix" value="CF1" />
      <param name="enable_logging_battery" value="True" />
    </node>
  </group>

  <group ns="CF2" if="$(eval 'CF2' in cf_names_arg)">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri2)" />
      <param name="tf_prefix" value="CF2" />
      <param name="enable_logging_battery" value="True" />
    </node>
  </group>

  <group ns="CF3" if="$(eval 'CF3' in cf_names_arg)">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri3)" />
      <param name="tf_prefix" value="CF3" />
      <param name="enable_logging_battery" value="True" />
    </node>
  </group>

  <group ns="CF4" if="$(eval 'CF4' in cf_names_arg)">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri4)" />
      <param name="tf_prefix" value="CF4" />
      <param name="enable_logging_battery" value="True" />
    </node>
  </group>

  <group ns="CF5" if="$(eval 'CF5' in cf_names_arg)">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri5)" />
      <param name="tf_prefix" value="CF5" />
      <param name="enable_logging_battery" value="True" />
    </node>
  </group>-->

  <!-- Node to send external positions from Qualisys -->
  <!--<node name="cf_qualisys_node" pkg="crazyflie_scripts" type="qualisys_cf_stream.py" output="screen">
    <param name="ip" value="$(arg ip)" />
  </node>-->

  <!-- Node for Crazyflie swarm controller -->
  <node name="run_crazyflies" pkg="crazyflie_scripts" type="run_crazyflies.py" output="screen">
  </node>

  <!-- Node for tracking charging pads -->
  <!--<node name="charge_qualisys_node" pkg="crazyflie_scripts" type="qualisys_charging_pad_stream.py" output="screen">
    <param name="ip" value="$(arg ip)" />
  </node>-->

</launch>
